messaging/src/main/java/org/axonframework/messaging/QualifiedName.java

```diff
@@ -0,0 +1,183 @@
+     * {@link QualifiedName#toSimpleString()}.
+     */
+    @SuppressWarnings("DuplicateExpressions")
+    static QualifiedName simpleStringName(@Nonnull String simpleString) {
+        assertNonEmpty(simpleString, "Cannot construct a QualifiedName based on a null or empty String.");
+        int bracketOffset = 1;
+        int signAndBracketOffset = 2;
+
+        String namespace = "";
+        String localName;
+        String revision = "";
+        int lastAt = simpleString.lastIndexOf('@');
+        int lastHash = simpleString.lastIndexOf('#');
+        int length = simpleString.length();
+
+        if (onlyHasLocalName(lastAt, lastHash)) {
+            localName = simpleString;
+        } else if (hasLocalNameAndNamespace(lastAt, lastHash)) {
+            localName = simpleString.substring(0, lastAt - bracketOffset);
+            namespace = simpleString.substring(lastAt + signAndBracketOffset, length - bracketOffset);
+        } else if (hasLocalNameAndRevision(lastAt, lastHash)) {
+            localName = simpleString.substring(0, lastHash - bracketOffset);
+            revision = simpleString.substring(lastHash + signAndBracketOffset, length - bracketOffset);
+        } else {
+            localName = simpleString.substring(0, lastAt - bracketOffset);
+            namespace = simpleString.substring(lastAt + signAndBracketOffset, lastHash - signAndBracketOffset);
+            revision = simpleString.substring(lastHash + signAndBracketOffset, length - bracketOffset);
+        }
+
+        return new SimpleQualifiedName(namespace, localName, revision);
+    }
+
+    private static boolean hasLocalNameAndRevision(int lastAt, int lastHash) {
+        return lastAt == -1 && lastHash != -1;
+    }
+
+    private static boolean hasLocalNameAndNamespace(int lastAt, int lastHash) {
+        return lastAt != -1 && lastHash == -1;
+    }
+
+    private static boolean onlyHasLocalName(int lastAt, int lastHash) {
+        return lastAt == -1 && lastHash == -1;
+    }
+
+    /**
+     * Returns the namespace this {@link QualifiedName} refers too.
+     * <p>
+     * The namespace may represent a (bounded) context, package, or whatever other "space" this name applies too.
+     *
+     * @return The namespace this {@link QualifiedName} refers too.
+     */
+    @Nonnull
+    String namespace();
+
+    /**
+     * Returns the local name this {@link QualifiedName} refers too.
+     * <p>
+     * The local name may represent a {@link Message Message's} name, a {@link MessageHandler} its name, or whatever
+     * other business specific concept this name should represent.
+     *
+     * @return The local name this {@link QualifiedName} refers too.
+     */
+    @Nonnull
+    String localName();
+
+    /**
+     * Returns the revision of this {@link QualifiedName}.
+     *
+     * @return The revision of this {@link QualifiedName}.
+     */
+    @Nonnull
+    String revision();
```

Comment:
Ah yes, that aligns more with how things are currently in place. Let's make this an `Optional`, indeed.
Making the getter return an `Optional` does mean we cannot use a `record`, though. It would impose that the field of the `revision` is an `Optional<String>` parameter throughout; something I'd rather avoid.